tests/tests_fabric/parity/test_parity_simple.py

```diff
@@ -0,0 +1,150 @@
+from tests_fabric.parity.models import ConvNet
+from tests_fabric.parity.utils import (
+    get_model_input_dtype,
+    is_cuda_memory_close,
+    is_state_dict_equal,
+    is_timing_close,
+    make_deterministic,
+)
+
+from lightning.fabric.fabric import Fabric
+
+
+def train_torch(
+    move_to_device: Callable,
+    precision_context,
+    input_dtype=torch.float32,
+):
+    make_deterministic()
+    memory_stats = {}
+
+    model = ConvNet()
+    model = move_to_device(model)
+    dataloader = model.get_dataloader()
+    optimizer = model.get_optimizer()
+    loss_fn = model.get_loss_function()
+
+    memory_stats["start"] = torch.cuda.memory_stats()
+
+    model.train()
+    iteration_timings = []
+    iterator = iter(dataloader)
+    for _ in range(model.num_steps):
+        t0 = time.perf_counter()
+
+        inputs, labels = next(iterator)
+        inputs, labels = move_to_device(inputs), move_to_device(labels)
+        optimizer.zero_grad()
+        with precision_context():
+            outputs = model(inputs.to(input_dtype))
+        loss = loss_fn(outputs.float(), labels)
+        loss.backward()
+        optimizer.step()
+
+        t1 = time.perf_counter()
+        iteration_timings.append(t1 - t0)
+
+    memory_stats["end"] = torch.cuda.memory_stats()
+
+    return model.state_dict(), torch.tensor(iteration_timings), memory_stats
+
+
+def train_fabric(fabric):
+    make_deterministic()
+    memory_stats = {}
+
+    model = ConvNet()
+    initial_state_dict = deepcopy(model.state_dict())
+
+    optimizer = model.get_optimizer()
+    model, optimizer = fabric.setup(model, optimizer)
+
+    dataloader = model.get_dataloader()
+    dataloader = fabric.setup_dataloaders(dataloader)
+    loss_fn = model.get_loss_function()
+
+    memory_stats["start"] = torch.cuda.memory_stats()
+
+    model.train()
+    iteration_timings = []
+    iterator = iter(dataloader)
+    for _ in range(model.num_steps):
+        t0 = time.perf_counter()
+
+        inputs, labels = next(iterator)
+        optimizer.zero_grad()
+        outputs = model(inputs)
+        loss = loss_fn(outputs, labels)
+        fabric.backward(loss)
+        optimizer.step()
+
+        t1 = time.perf_counter()
+        iteration_timings.append(t1 - t0)
+
+    memory_stats["end"] = torch.cuda.memory_stats()
+
+    # check that the model has changed
+    assert not is_state_dict_equal(initial_state_dict, model.state_dict())
+
+    return model.state_dict(), torch.tensor(iteration_timings), memory_stats
+
+
+@pytest.mark.flaky(reruns=3)
+@pytest.mark.usefixtures("reset_deterministic_algorithm", "reset_cudnn_benchmark")
+@pytest.mark.parametrize(
+    "precision, accelerator",
+    [
+        (32, "cpu"),
+        pytest.param(32, "gpu", marks=RunIf(min_cuda_gpus=1)),
+        # pytest.param(16, "gpu", marks=RunIf(min_cuda_gpus=1)),  # TODO: requires GradScaler
+        pytest.param("bf16", "cpu"),
+        pytest.param("bf16", "gpu", marks=RunIf(min_cuda_gpus=1, bf16_cuda=True)),
```

Comment:
Let's be explicit. Same suggestion elsewhere

```suggestion
        pytest.param(32, "cuda", marks=RunIf(min_cuda_gpus=1)),
        # pytest.param(16, "cuda", marks=RunIf(min_cuda_gpus=1)),  # TODO: requires GradScaler
        pytest.param("bf16", "cpu"),
        pytest.param("bf16", "cuda", marks=RunIf(min_cuda_gpus=1, bf16_cuda=True)),
```